src/metrics/metric_types.go

```diff
@@ -20,3 +24,61 @@ func (qd QueryDefinition) GetDataModels() interface{} {
 	ptr := reflect.New(reflect.ValueOf(qd.dataModels).Type())
 	return ptr.Interface()
 }
+
+func (qd *QueryDefinition) insertDatabaseNames(databases collection.DatabaseList) *QueryDefinition {
```

Comment:
is it unclear whether these functions will modify or not the instance calling it. If not we could change the signature not to use the pointer.

Reply:
Done